### PR TITLE
Various refactoring for easier readability

### PR DESCRIPTION
**Pull request by Ilithor**

### CODE REVIEW

File: src/components/post/postList/DeletePost.js [hunk 1]
```diff
@@ -43,17 +40,11 @@ const DeletePost = ({
...
-              Cancel
-            </Button>
-            <Button onClick={deletePost} color='secondary'>
-              Delete
-            </Button>
-          </DialogActions>
-        </Dialog>
+        <DeletePostDialog
+          open={open}
+          handleClose={handleClose}
```
Comment (jeffersoneagley) [inline src/components/post/postList/DeletePost.js]: Technically, you could make this component return the CustomButton when !open, and move all the state onboard. You would end up passing in nothing but deletePost, or even just passing in post and let it handle the delete itself too.

File: src/components/profile/user/editDetails/EditDetailsForm.js [hunk 2]
```diff
@@ -0,0 +1,60 @@
...
+
+// MUI
+import TextField from '@material-ui/core/TextField';
+import withStyles from '@material-ui/core/styles/withStyles';
+import style from '../../../../style/style';
+
+/** Displays the edit form to edit the user's details
+ * @type {React.FunctionComponent}
+ * @param {object} props
+ * @param {object} props.classes
```
[the remark this answers is not in the document]
Comment (Ilithor) [reply src/components/profile/user/editDetails/EditDetailsForm.js]: I don't think so. It just contains properties that help MUI know what styles to apply.

File: src/components/post/postList/DeletePost.js (hunk 1 above)
Comment (Ilithor) [reply src/components/post/postList/DeletePost.js]: My only issue is that would increase the number of indentations in the component, which was the reason for the refactor.

Comment (jeffersoneagley) [reply src/components/post/postList/DeletePost.js]: No I mean you can move the delete button and the "open" state barracks into your DeletePostDialog component. Just adding a usestate hook and then
if(!open){
Return custom button here
}
Before you return the actual modal.

File: src/components/profile/user/editDetails/EditDetailsForm.js (hunk 2 above)
[the remark this answers is not in the document]
Comment (jeffersoneagley) [reply src/components/profile/user/editDetails/EditDetailsForm.js]: See if MUI has a type listed in their docs

File: src/components/post/postList/DeletePost.js (hunk 1 above)
Comment (jeffersoneagley) [reply src/components/post/postList/DeletePost.js]: Back, not barracks

File: src/components/profile/user/profile/userProfileDisplay.test.js [hunk 3]
```diff
@@ -4,8 +4,8 @@ import { mount } from 'enzyme';
 import ConnectedUserProfileDisplay, {
   // eslint-disable-next-line no-unused-vars
   IUserProfileDisplayComponentProps,
-} from './UserProfileDisplay';
-import TestWithReduxStore from '../../../fixtures/component/TestWithReduxStore';
+} from './index';
```
[the remark this answers is not in the document]
Comment (Ilithor) [reply src/components/profile/user/profile/userProfileDisplay.test.js]: Fixed

File: src/components/post/postList/DeletePost.js (hunk 1 above)
Comment (Ilithor) [reply src/components/post/postList/DeletePost.js]: Fixed

**Pull request merged**